site_scons/daos_build.py

```diff
@@ -86,6 +88,10 @@ def _known_deps(env, **kwargs):
         libs = set(env.get('LIBS', []))
 
     known_libs = libs.intersection(set(libraries.keys()))
+    missing_libs = libs - known_libs
+    for item in missing_libs:
+        if item not in missing:
+            missing.add(item)
```

Comment:
The clever thing about sets is you don't need to check before you insert it.
```suggestion
        missing.add(item)
```
but there's something bigger here as well, you're working with sets so I don't think you need to the for loop either.

missing.update(libs - known_libs)